src/main/java/org/commcare/formplayer/services/CaseSearchHelper.java

```diff
@@ -129,10 +129,12 @@ private void parseIntoCaseSearchStorage(SQLiteDB caseSearchDb, UserSqlSandbox ca
             XmlPullParserException, IOException {
         try {
             DbUtils.setAutoCommit(caseSearchDb, false);
+            caseSearchIndexTable.delete();
             caseSearchIndexTable.createTable();
             CaseInstanceXmlTransactionParserFactory factory = new CaseInstanceXmlTransactionParserFactory(
                     caseSearchSandbox, caseSearchIndexTable);
             caseSearchStorage.initStorage();
+            caseSearchStorage.removeAll();
```

Comment:
This implies that there may be data in the table immediately after `initStorage`. Is this because case data may persist in an `ExternalDataInstance`?

Reply:
no, `initStorage` is called optimistically to ensure that the tables are created. Ideally it would return a boolean indicating whether the tables were created on not which would allow us to skip the `removeAll` call for fresh tables.